### PR TITLE
Genargs are not holes

### PR DESCRIPTION
**Pull request by SkySkimmer**
ie split CHole/GHole to have a separate case for genargs

Overlays:
- https://github.com/coq-community/bignums/pull/72
- https://github.com/mattam82/Coq-Equations/pull/537
- https://github.com/LPCIC/coq-elpi/pull/432
- https://github.com/QuickChick/QuickChick/pull/316
- https://github.com/MetaCoq/metacoq/pull/835

### CODE REVIEW

**Review by gares — APPROVED** [review]

File: test-suite/output/bug_7443.out [hunk 1]
```diff
@@ -1,7 +1,13 @@
 Literal 1
      : Type
+File "./output/bug_7443.v", line 21, characters 2-46:
+Warning: Notation "[ _ ]" was already used in scope foo_scope.
+[notation-overridden,parsing]
```
Comment (SkySkimmer) [inline test-suite/output/bug_7443.out:5]: This warning seems reasonable as we are overriding the notation, does anyone disagree?

Comment (herbelin) [reply test-suite/output/bug_7443.out:5]: Looks expected, indeed. So, wasn't it issued because of the `ltac:(...)`?

**Comment by herbelin** [comment]
Looks well clearer.

File: test-suite/output/bug_7443.out (hunk 1 above)
Comment (SkySkimmer) [reply test-suite/output/bug_7443.out:5]: The notation comparison saw `(Literal (t:=reify_type_of x) x)` ie `(@Literal (reify_type_of x) x)` as `(@Literal _ x)` so equal to the second notation `(Literal x)`
Now genarg and hole don't compare equal so it sees a difference.

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Review by ppedrot — APPROVED** [review]
Looks superficially good but this kind of change is always tricky.

**Comment by SkySkimmer** [comment]
Ready to merge

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**